WordPress/src/main/java/org/wordpress/android/ui/accounts/login/Login2FaFragment.java

```diff
@@ -274,13 +356,62 @@ public void onAuthenticationChanged(OnAuthenticationChanged event) {
 
         AppLog.i(T.NUX, "onAuthenticationChanged: " + event.toString());
 
-        doFinishLogin();
+        if (isSocialLoginConnect) {
+            AccountStore.PushSocialLoginPayload payload = new AccountStore.PushSocialLoginPayload(mIdToken, mService);
+            mDispatcher.dispatch(AccountActionBuilder.newPushSocialConnectAction(payload));
+        } else {
+            doFinishLogin();
+        }
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onSocialChanged(OnSocialChanged event) {
+        if (event.isError()) {
+            switch (event.error.type) {
+                // Two-factor authentication code was incorrect; save new nonce for another try.
+                case INVALID_TWO_STEP_CODE:
+                    endProgress();
+
+                    switch (mType) {
```

Comment:
I hinted in an earlier comment about this so, I think we should just deduce the type from the `event.error.nonce` without relying on a global instance variable, unless of course there is more to it. Is there some special reason to rely on a global var here?

Reply:
As I mentioned in my reply, we can't deduce the type from the nonce.